tests/test_init.py

```diff
@@ -26,3 +26,208 @@ def test_systemd_present(container):
     assert container.connection.exists("systemctl")
     assert container.connection.file("/etc/machine-id").exists
     assert container.connection.run_expect([0], "systemctl status")
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_boottime(container):
+    """Check that the startup time of the container is reasonable"""
+
+    # internal helper function to extract the time from systemd-analyze time
+    # extracts a timestamp with a given prefix and converts it to seconds
+    def extract_time(stdout, prefix):
```

Comment:
I would also suggest to include the expected form of the output of systemd-analyze and explain how and what you are extracting.
```suggestion
    def extract_time(stdout, prefix):
        """internal helper function to extract the time from systemd-analyze time
        extracts a timestamp with a given prefix and converts it to seconds"""
```

Reply:
I added an output example and described how this function is expected to behave. It is better now, but still reads a bit bulky to me, if you have a better suggestion feel free to comment.